src/value.rs

```diff
@@ -139,6 +139,24 @@ impl Value {
         }
     }
 
+    /// Tries to downcast to a `&TypedValue`.
+    ///
+    /// Returns `Some(&TypedValue<T>)` if the value carries a type corresponding
+    /// to `T` and `None` otherwise.
+    pub fn downcast_ref<'a, T: FromValueOptional<'a> + SetValue>(&self) -> Option<&TypedValue<T>> {
+        unsafe {
+            let ok = from_glib(
+                gobject_ffi::g_type_check_value_holds(mut_override(self.to_glib_none().0),
+                    T::static_type().to_glib()));
+            if ok {
+                Some(mem::transmute(self))
```

Comment:
The value transmutes are safe because the only difference between `Value` and `TypedValue` is the zero-sized phantom data.

Reply:
This comment not better be in code? ;)